### PR TITLE
feature: set prettier formatter to pre-commit hook

### PR DESCRIPTION
**Pull request by yomybaby**
## ⚠️
Please run `npm install` at the root of this project. After that, the pre-commit hook will work with prettier.

---

### Summary
🎨🐛🚀

This pull request adds tools and scripts to enforce code style and formatting rules using `husky`, `lint-staged`, and `prettier`. It also applies the prettier configuration to various files in the `react/src/components` and `react/src` folders, and fixes some minor syntax errors and inconsistencies.

> _Sing, O Muse, of the code that was polished and refined_
> _By the skillful developers, who sought to align_
> _Their style and their syntax with `prettier`'s rules_
> _And to fix all the errors with `lint-staged`'s tools_

### Walkthrough
*  Add and configure tools to format and lint code before committing ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L31-R32),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L127-R134),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R158-R170),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d2bc4bbf14eadc292d84e0ef5f7a93115c23557f533809fc80b896934291529dR1-R4),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-b640b344ee7f3f03d2a443795a5d0708ef50e2e6e34214109ab2aad13ad6ba98R2-R4),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-979a4a52db5287c891eac28a141bfa97bf8b0732ed5afae848d90956ef80e78dR1-R12),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R104))
*  Sort and group import statements according to the prettier configuration in various files ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-47bd17d906c2c7eabc2ddc244c9dff34ed49a7a0f97eb0c8cf787733a068a49cL1-R4),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-07eb2e622929dfdf4a7951b06bf97007b92191fb855dd16193961dec06c5eee1L1-R2),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860R1-R11),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L1-R12),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-4b96151459a8f1696be2142f66ea0a17ab2076109483b7ecc4498175e0efbaebL1-R8),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-463d993a4d8752e78427abdcd400b131ca6e3bb205434169a3b17c61a8637b7cL1-R4),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d1ec752f80f23409a9e773447a81836f8e37cf5a486f84a53bd648dcdd92fe21L1-R5),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-87926a934bdbef7acc81ac28c98c5fb9f55a106c41475b24da9509531dfe3c96L1-R9),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-09191658ed8719d26840abbd870770e8ba9d388ea32bf8a9f30f47a498707a95L1-R19),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-9e73d2f3e747ee272e25e7891fcc0581a566030054eb32ef425cb3859f433e44L1-R4),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L1-R18),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7f2c06444d9aa6ff412a301084235fff8fee780cac6bce18aa2e9b0642d22becL1-R2),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL1-R5),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L1-R7),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L1-R12))
*  Replace double quotes with single quotes in various files according to the prettier configuration ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L57-R59),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L70-R68),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L85-R87),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L111-R109),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L133-R131),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L166-R167),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L23-R29),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L36-R39),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L45-R53),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-4b96151459a8f1696be2142f66ea0a17ab2076109483b7ecc4498175e0efbaebL29-R29),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-4b96151459a8f1696be2142f66ea0a17ab2076109483b7ecc4498175e0efbaebL37-R36),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-463d993a4d8752e78427abdcd400b131ca6e3bb205434169a3b17c61a8637b7cL19-R19),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-463d993a4d8752e78427abdcd400b131ca6e3bb205434169a3b17c61a8637b7cL25-R26),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d1ec752f80f23409a9e773447a81836f8e37cf5a486f84a53bd648dcdd92fe21L20-R25),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-87926a934bdbef7acc81ac28c98c5fb9f55a106c41475b24da9509531dfe3c96L17-R23),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-87926a934bdbef7acc81ac28c98c5fb9f55a106c41475b24da9509531dfe3c96L35-R39),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-09191658ed8719d26840abbd870770e8ba9d388ea32bf8a9f30f47a498707a95L31-R41),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-09191658ed8719d26840abbd870770e8ba9d388ea32bf8a9f30f47a498707a95L51-R51),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-09191658ed8719d26840abbd870770e8ba9d388ea32bf8a9f30f47a498707a95L62-R79),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-9e73d2f3e747ee272e25e7891fcc0581a566030054eb32ef425cb3859f433e44L12-R12),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L70-R73),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L119-R119),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L136-R136),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L160-R161),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L180-R188),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L204-R206),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L231-R231),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L238-R239),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L266-R266),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L274-R275),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L281-R281),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L295-R295),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L324-R324),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L330-R330),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L341-R341),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L361-R369),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L375-R385),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L406-R406),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L413-R413),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L427-R427),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L438-R438),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L444-R446),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L465-R465),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L472-R472),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7f2c06444d9aa6ff412a301084235fff8fee780cac6bce18aa2e9b0642d22becL9-R9),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7f2c06444d9aa6ff412a301084235fff8fee780cac6bce18aa2e9b0642d22becL16-R17),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL57-R57),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL63-R70),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL89-R93),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL99-R99),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL110-R110),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL122-R122),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL132-R132),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL139-R141),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL150-R150),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL156-R164),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL174-R174),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL181-R216),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL223-R223),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL237-R238),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL245-R245),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL251-R253),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL259-R261),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL267-R268),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL277-R278),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-2e4093533b8ad67f24c5293283b509735a1a4ec17a44ec258a25d6a44e82337dL287-R288),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L19-R19),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L28-R28),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L35-R46),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L56-R61),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L69-R77),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L86-R91),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L98-R98),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-5186c0abf1b634ad9a817eef960fe0d6762c0d3415fc89bb9742b7dd21867368L109-R109),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L49-R48),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L64-R63),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L96-R95),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L102-R101),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L110-R113))
*  Add commas after some parameters and function calls according to the prettier configuration ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L7-R26),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L50-R51),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L107-R108),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L198-R198),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L266-R266),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L295-R295),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0c3eda30e966d59fcc7f298b411bdf4671271d6ff721e2395f9985d638a98e37L465-R465),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-e1757ecad7e9cf3ff6f03bb9ab94ca5fb3693ed22cfdccada07d28f46c46028aL17-R17),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-e1757ecad7e9cf3ff6f03bb9ab94ca5fb3693ed22cfdccada07d28f46c46028aL23-R24),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-e1757ecad7e9cf3ff6f03bb9ab94ca5fb3693ed22cfdccada07d28f46c46028aL34-R36),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-e1757ecad7e9cf3ff6f03bb9ab94ca5fb3693ed22cfdccada07d28f46c46028aL45-R45),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L40-R39),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d51185cead75f0b0dcc37cd5a43ecb02515f223126c10b69836995f207cbe1a1L49-R48))
*  Remove an extra space in the test:react script in `package.json` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L13-R13))
*  Move the ReactWebComponentProps interface definition from line 16 to line 11 in `DefaultProviders.tsx` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-1ec6f89edfe2d9ae74244de78c2c81173269c809f471dd0195a10bc02420f860L7-R26))
*  Add a .prettierrc.json file to configure the prettier options ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-979a4a52db5287c891eac28a141bfa97bf8b0732ed5afae848d90956ef80e78dR1-R12))
*  Add a dependency on @trivago/prettier-plugin-sort-imports in `package.json` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R104))
*  Add a script to run lint-staged before committing in `.husky/pre-commit` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-d2bc4bbf14eadc292d84e0ef5f7a93115c23557f533809fc80b896934291529dR1-R4))
*  Update the .prettierignore file to exclude some files and folders from formatting ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-b640b344ee7f3f03d2a443795a5d0708ef50e2e6e34214109ab2aad13ad6ba98R2-R4))
*  Add a helper function to modify GraphQL queries according to the availability of a supported field in `RelayEnvironment.ts` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-e1757ecad7e9cf3ff6f03bb9ab94ca5fb3693ed22cfdccada07d28f46c46028aR1-R2))
*  Remove an unused import statement for createClient in `RelayEnvironment.ts` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-e1757ecad7e9cf3ff6f03bb9ab94ca5fb3693ed22cfdccada07d28f46c46028aL9-R11))
*  Add some tests for the DomainSelector and Flex components in `DomainSelector.test.tsx` and `Flex.test.tsx` ([link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L1-R12),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L23-R29),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L36-R39),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-0cfc6cf757b6fbea081da2183561c79b213b428b2069f2e9d76852131ecd55e8L45-R53),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-87926a934bdbef7acc81ac28c98c5fb9f55a106c41475b24da9509531dfe3c96L1-R9),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-87926a934bdbef7acc81ac28c98c5fb9f55a106c41475b24da9509531dfe3c96L17-R23),[link](https://github.com/lablup/backend.ai-webui/pull/1884/files?diff=unified&w=0#diff-87926a934bdbef7acc81ac28c98c5fb9f55a106c41475b24da9509531dfe3c96L35-R39))

### CODE REVIEW

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "trailingComma": "es5",
```
Comment (agatha197) [inline .prettierrc.json]:
```suggestion
```

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "tabWidth": 2,
+  "semi": true,
+  "singleQuote": true,
+  "importOrderSeparation": true,
+  "importOrderParserPlugins": ["typescript", "jsx"],
+  "printWidth": 80,
+  "bracketSpacing": true,
+  "trailingComma": "all",
```
Comment (agatha197) [inline .prettierrc.json:9]: Duplicate key with line 2.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]
Oh I got to know there is a duplicated key on the .prettierrc.json

File: src/.prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "trailingComma": "es5",
```
Comment (agatha197) [inline src/.prettierrc.json]:
```suggestion
```

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]

File: .prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "plugins": ["@trivago/prettier-plugin-sort-imports"]
```
Comment (agatha197) [inline .prettierrc.json:10]: Possibly needed here, too.
```suggestion
  "plugins": ["@trivago/prettier-plugin-sort-imports"]
  "htmlWhitespaceSensitivity": "ignore"
```

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]

File: .prettierrc.json (hunk 4 above)
Comment (yomybaby) [reply .prettierrc.json:10]: Thank you for mentioning that. I have searched for this option again and confirmed that it doesn't affect JSX code. It only affects [HTML template literals in JavaScript](https://prettier.io/blog/2018/11/07/1.15.0#html-template-literal-in-javascript). Therefore, we might not need to configure this option at the root level (for React).

**Review by Sujin-Kim1 — APPROVED** [review]

File: src/components/backend-ai-import-view.ts [hunk 5]
```diff
@@ -461,30 +535,37 @@
...
-      if (err && err.message) {
-        this.notification.text = PainKiller.relieve(err.title);
-        this.notification.detail = err.message;
-        this.notification.show(true, err);
-      }
-    });
+    return globalThis.backendaiclient.vfolder
+      .create(name, host, group, usageMode, permission)
+      .then((value) => {
+        if (url.includes('github.com/')) {
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply src/components/backend-ai-import-view.ts:545]: I'll handle this in a separated PR.

**Pull request merged**